# Importing Packages and Libraries

In [1]:
import numpy as np
import os
import pandas as pd
import cv2
import matplotlib.pyplot as plt
from tqdm import tqdm

# Defining Some Paths

In [2]:
Path_train = os.path.join("./train/images")
Path_val = os.path.join("./valid/images")
Path_test = os.path.join("./test/images")

# Appending Images to lists to view shapes, length etc

In [18]:
Common_File_Shapes = []
for idxFile, File in tqdm(enumerate(os.listdir(Path_train))):
    File_Content = cv2.imread(os.path.join(Path_train, File))
    Shape = np.shape(File_Content)
    Common_File_Shapes.append(Shape)

for idxFile, File in tqdm(enumerate(os.listdir(Path_val))):   
    File_Content = cv2.imread(os.path.join(Path_val, File))
    Shape = np.shape(File_Content)
    Common_File_Shapes.append(Shape)

for idxFile, File in tqdm(enumerate(os.listdir(Path_test))):   
    File_Content = cv2.imread(os.path.join(Path_test, File))
    Shape = np.shape(File_Content)
    Common_File_Shapes.append(Shape)

4200it [00:01, 2182.78it/s]
1704it [00:00, 2203.43it/s]
100it [00:00, 2214.85it/s]


Total Image Count = 6004, 4200 in train, 1704 in val, 100 in test

In [19]:
print(f'Total Images Count: {len(Common_File_Shapes)}')

Total Images Count: 6004


In [22]:
print(np.unique(Common_File_Shapes))

[  3 416]


In [25]:
images_width, images_height = Common_File_Shapes[0][0], Common_File_Shapes[0][1]
print(f'Width: {images_width}, Height: {images_height}')

Width: 416, Height: 416
